### ISSUE TITLE
Introduction of log levels 

### ISSUE THREAD

**Issue by markusschloesser** [markusschloesser/MackieC4_P3#105]
Current situation:
Either log everything that has a log entry or nothing. Especially for broader releases that causes issues. We don't want to have to much logging in a release. But adding and removing or out commenting is time consuming.

Proposal:
Introduce log levels similar to other dev projects with debug, warning, error, information etc
And then decide in ONE line (in MackieC4.py), which log level is the active one

To do:
Each and every single log line needs to have the level in it as parameter

**Comment by BilldarBagdar** [comment]
There are "log levels" in effect for Live already. (If you look at InputControlElement.report_values() method, you'll see it logs to "debug level", and if you look at the log_message() method in ControlSurface, you'll see it logs to "info level".  Unfortunately, we in the public never see "debug level" messages because our "release version" of Live only ever runs at "info" log level)

But I get what you mean, for sure.   Again, I think that kind of thing could be done, but is it worth it?  Would log messages we "clean up" because the problem got solved ever be needed again anyway?  With git, they always still exist in the repo history.

**Comment by markusschloesser** [comment]
I meant C4 script log levels (but wrote on the phone, so lazy and brief). The thing is, we now have a whole lot of logging going on,  to the amount that it makes following the log file really hard. So we need to scale down logging anyway, which means "touching" all log code lines anyway. But if there's ever a problem again with devices or tracks, or I find I want to implement new things, we'll need the logging again. So why not touch all the lines now ones and then never again, and just up the level in main? Maybe also with categories like tracks, devices etc.

**Comment by BilldarBagdar** [comment]
Yes, I understand.  It might be a fun "python exercise" for you.  What I think you are describing would best be implemented by making a top-level "log message" method that also takes the extra "log level" parameter and then only "really logs" the message when the "log level" is correct at runtime.  (This technique won't be very efficient because the new logger method still gets called from everywhere at runtime, but you could control the script's final logging verbosity as desired using it.)

So, when you are going through all the code and updating every current log line, you end up just editing the logger method call a little to call the new logging method at an appropriate "log level".  

What I mean by "fun" is, this seems like a great opportunity for a "lazy programmer" to write a little "python script" that can do the tedious task of updating every logger line in the project.  Such a Python script would "walk" the project folder content (the project files) and substitute the "new logger call" for the "old logger call" everywhere it finds an "old logger call".  The script would probably need to leverage at least one "regular expression" to identify candidate lines to replace (or make substitutions in)

A good "lazy programmer" will write a script to, for example, add an import statement to every project file rather than opening every file and "manually" adding the import, even if when writing such a script takes longer than opening every file or is "hard to script correctly".  The "lazy way" is to write a script so the task can be (automated) performed over and over again forever just by running the script again.
